pkg/chunk/cassandra/storage_client.go

```diff
@@ -98,13 +103,19 @@ func (cfg *Config) session() (*gocql.Session, error) {
 	cluster.ConnectTimeout = cfg.ConnectTimeout
 	cluster.ReconnectInterval = cfg.ReconnectInterval
 	cluster.NumConns = cfg.NumConnections
+	cluster.Logger = log.With(pkgutil.Logger, "module", "gocql", "client", name)
+	cluster.Registerer = prometheus.WrapRegistererWith(
+		prometheus.Labels{"client": name}, prometheus.DefaultRegisterer)
 	if cfg.Retries > 0 {
 		cluster.RetryPolicy = &gocql.ExponentialBackoffRetryPolicy{
 			NumRetries: cfg.Retries,
 			Min:        cfg.MinBackoff,
 			Max:        cfg.MaxBackoff,
 		}
 	}
+	if cfg.ConvictHosts {
```

Comment:
Should this be `!cfg.ConvictHosts`.  i.e. if the user does not want to convict hosts then we replace the `cluster.ConvictionPolicy` with noop.

This would also maintain current behavior with default values.